packages/kit/src/runtime/client/router.js

```diff
@@ -244,6 +283,24 @@ export class Router {
 		return this.renderer.load(info);
 	}
 
+	/** @param {(navigationIntent: import('./types').NavigationIntent) => void} fn */
+	onBeforeNavigate(fn) {
+		/** @param {Event} event*/
+		function onBeforeNavEventListener(event) {
+			/** @type {any}*/
```

Comment:
it'd probably be better to create a type for this than to use `any`
```suggestion
			/** @type {any} */
```

Reply:
You are right, I've added a CustomEvent type cast here